packages/docusaurus-plugin-content-blog/src/authors.ts

```diff
@@ -68,17 +69,26 @@ export async function getAuthorsMap(params: {
 type AuthorsParam = {
   frontMatter: BlogPostFrontMatter;
   authorsMap: AuthorsMap | undefined;
+  baseUrl: string;
 };
 
 // Legacy v1/early-v2 front matter fields
 // We may want to deprecate those in favor of using only frontMatter.authors
-function getFrontMatterAuthorLegacy(
-  frontMatter: BlogPostFrontMatter,
-): Author | undefined {
+function getFrontMatterAuthorLegacy({
+  baseUrl,
+  frontMatter,
+}: {
+  baseUrl: string;
+  frontMatter: BlogPostFrontMatter;
+}): Author | undefined {
   const name = frontMatter.author;
   const title = frontMatter.author_title ?? frontMatter.authorTitle;
   const url = frontMatter.author_url ?? frontMatter.authorURL;
-  const imageURL = frontMatter.author_image_url ?? frontMatter.authorImageURL;
+  let imageURL = frontMatter.author_image_url ?? frontMatter.authorImageURL;
+
+  if (baseUrl !== '/' && imageURL && !imageURL.startsWith('http')) {
+    imageURL = path.join(baseUrl, imageURL);
+  }
```

Comment:
duplicated logic, encapsulate in shared function

Also I think we should process all urls starting with `/`, not based on protocol. 
We should not process relative urls like `./img/author.png`

I'd use something like:

```js
function normalizeImageURL({imageURL,baseUrl}) {
  return imageURL.startsWith("/") ? normalizeUrl([baseUrl,imageURL)) : imageUrl
}
```